File: src/main/java/com/devon/dapuzzo/CustomErrorController.java
package com.devon.dapuzzo;

import org.springframework.boot.autoconfigure.web.ErrorController;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

/**
 * Created by devondapuzzo on 8/20/17.
 */
@Controller
public class CustomErrorController implements ErrorController {

    private static final String PATH = "/error";

    @RequestMapping(value = PATH)
    public String error() {
        return "index";
    }

    @Override
    public String getErrorPath() {
        return PATH;
    }
}

File: src/main/java/com/devon/dapuzzo/CustomErrorController.kt
package com.devon.dapuzzo

import org.springframework.boot.autoconfigure.web.ErrorController
import org.springframework.web.bind.annotation.RequestMapping

/**
 * Created by devondapuzzo on 8/20/17.
 */
//@Controller
class CustomErrorController : ErrorController {

    @RequestMapping(value = PATH)
    fun error(): String {
        return "index"
    }

    override fun getErrorPath(): String {
        return PATH
    }

    companion object {
        const private val PATH = "/error"
    }
}
